### ISSUE TITLE
android compile issue

### ISSUE THREAD

**Issue by fayfive** [libsdl-org/SDL#5354]
I follow the step of the doc.

./androidbuild.sh org.libsdl.testgles ../test/testgles.c
cd /usr/src/SDL2/build/org.libsdl.testgles/
./gradlew installDebug

fatal error: 'dbus/dbus.h' file not found

SDL2.0.20
NDK 17c
I also try ndk 21.3

**Comment by slouken** [comment]
Can you include the full compile output? Android shouldn't be including dbus.h anywhere.

**Comment by fayfive** [comment]
Here is the log of "ANDROID_NDK=/home/alfred/Android/Sdk/ndk/21.3.6528147"
```
> Task :app:buildNdkBuildDebug FAILED
C/C++: In file included from /home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/jni/SDL/src/SDL.c:38:
C/C++: /home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/jni/SDL/src/core/linux/SDL_dbus.h:30:10: fatal error: 'dbus/dbus.h' file not found
C/C++: #include <dbus/dbus.h>
C/C++:          ^~~~~~~~~~~~~
C/C++: 1 error generated.
C/C++: make: *** [/home/alfred/Android/Sdk/ndk/21.4.7075529/build/core/build-binary.mk:476: /home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/build/intermediates/cxx/Debug/t3t2a4bd/obj/local/armeabi-v7a/objs-debug/SDL2/src/SDL.o] Error 1

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:buildNdkBuildDebug'.
> Build command failed.
  Error while executing process /home/alfred/Android/Sdk/ndk/21.4.7075529/ndk-build with arguments {NDK_PROJECT_PATH=null APP_BUILD_SCRIPT=/home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/jni/Android.mk NDK_APPLICATION_MK=/home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/jni/Application.mk APP_ABI=armeabi-v7a NDK_ALL_ABIS=armeabi-v7a NDK_DEBUG=1 NDK_OUT=/home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/build/intermediates/cxx/Debug/t3t2a4bd/obj NDK_LIBS_OUT=/home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/build/intermediates/cxx/Debug/t3t2a4bd/lib APP_PLATFORM=android-16 SDL2 main}
  [armeabi-v7a] Compile thumb  : SDL2 <= SDL_error.c
  [armeabi-v7a] Compile thumb  : SDL2 <= SDL_log.c
  [armeabi-v7a] Compile thumb  : SDL2 <= SDL_assert.c
  [armeabi-v7a] Compile thumb  : SDL2 <= SDL_dataqueue.c
  [armeabi-v7a] Compile thumb  : SDL2 <= SDL.c
  
  In file included from /home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/jni/SDL/src/SDL.c:38:
  /home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/jni/SDL/src/core/linux/SDL_dbus.h:30:10: fatal error: 'dbus/dbus.h' file not found
  #include <dbus/dbus.h>
           ^~~~~~~~~~~~~
  1 error generated.
  make: *** [/home/alfred/Android/Sdk/ndk/21.4.7075529/build/core/build-binary.mk:476: /home/alfred/working/SDL2-2.0.20/build/org.libsdl.testgles/app/build/intermediates/cxx/Debug/t3t2a4bd/obj/local/armeabi-v7a/objs-debug/SDL2/src/SDL.o] Error 1


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
19 actionable tasks: 19 executed

```

**Comment by fayfive** [comment]
> For some reason HAVE_DBUS_DBUS_H is enabled. Have you previously built SDL for Linux in this same source tree? I would suggest getting a fresh copy of SDL and seeing if you still have the issue.

You are right! I have unzip a new source. It works well.
I have built SDL for Linux in this same source tree.
But I have done "make distclean".
Dose it clean all the config?
